hal/src/clocks/mod.rs

```diff
@@ -28,7 +28,7 @@ For example, if we want to configure the [`MainClock`]:
 use atsamx7x_hal as hal;
 use hal::fugit::RateExtU32;
 
-let pac = hal::pac::Peripherals::take().unwrap();
+let pac = unsafe{hal::pac::Peripherals::steal()};
```

Comment:
It's `unsafe` again??? 🤔

Reply:
`steal()` is unsafe as it returns a peripheral struct without checking if the peripherals have been generated before, potentially violating the singleton assumption.

It seems `take()` is a method implemented for iterators which confuses the doc test since Peripherals is not an iterator.

Reply:
And yes, I believe you are correct that `critical-section` is the root cause here.

Reply:
Looking into this a bit more and rolling back to an older version of the pac makes this error go away.

Reply:
It seems `Peripherals` does not have the `take()` method anymore. Very unclear, investigating further.

Reply:
This is interesting. I should have some time tomorrow to branch and try next version of `svd2rust` to see if it's intermittent problem. There's 0.30.1 and two other point releases to try with.

I'm not going straight to last version as there are breaking changes and invocation changes + I want something to work on release automation (I wish I have like two weeks full time to work on that).

Reply:
Sounds good, I will keep investigating in the meantime.
I can't seem to build the pacs locally. `python3 tools/pacs.py svd` should work, right?

Reply:
https://github.com/rust-embedded/svd2rust/issues/704 Found the relevant issue.

Reply:
we don't catch this in our examples as they all utse RTIC and RTIC does a `steal()` internally.

Reply:
`tools/pacs.py` requires mapping JSON file as argument... I forgot to commit it - harvester generates it when downloads atpacks and extract SVDs.
For now you can run harvester yourself to redownload and obtain the file. It's used to fill up ATPACKs version numbers in `README.md`s of each PAC.